### PR TITLE
switch to std::thread

### PR DESCRIPTION
**Pull request by dvrogozh**

### CODE REVIEW

File: _studio/mfx_lib/scheduler/src/mfx_scheduler_core.cpp [hunk 1]
```diff
@@ -82,20 +81,24 @@ mfxSchedulerCore::~mfxSchedulerCore(void)
...
-        vm_thread_linux_schedparams params;
+        if (handle.joinable()) {
+            struct sched_param param{};
 
-        memset(&params, 0, sizeof(params));
-        params.schedtype = m_param.params.SchedulingType;
-        params.priority = m_param.params.Priority;
-        return vm_thread_set_scheduling(&handle, &params);
+            param.sched_priority = m_param.params.Priority;
+            return !pthread_setschedparam(handle.native_handle(), m_param.params.SchedulingType, &param);
```
Comment (alariono) [inline _studio/mfx_lib/scheduler/src/mfx_scheduler_core.cpp]: This is not part of std::thread...
So, MSDK still keep dependency from pthread and code is not cross platform

Comment (mgonchar) [reply _studio/mfx_lib/scheduler/src/mfx_scheduler_core.cpp]: @alariono I thought the intent of this change not to make code cross-platform, but to get rid of own thread wrapper

Comment (dvrogozh) [reply _studio/mfx_lib/scheduler/src/mfx_scheduler_core.cpp]: @mgonchar : exactly. std::thread as all std library don't support everything in the world. Priorities operations are still missing there and should be handle thru native handles. They even have the corresponding example: https://en.cppreference.com/w/cpp/thread/thread/native_handle. The major idea is to remove our wrapper or rather remove VM wrapper and if we really need any wrapper move it to the mediasdk core. In this specific case I don't believe we will need the above operation anywhere else outside the scheduler, thus, this is implemented right in the scheduler, not as a wrapper inside mediasdk core.

Comment (mgonchar) [reply _studio/mfx_lib/scheduler/src/mfx_scheduler_core.cpp]: @dvrogozh I think next candidate for clean up is our mutexes/lockers

File: _studio/mfx_lib/scheduler/src/mfx_scheduler_core.cpp [hunk 2]
```diff
@@ -82,20 +81,24 @@ mfxSchedulerCore::~mfxSchedulerCore(void)
 
 } // mfxSchedulerCore::~mfxSchedulerCore(void)
 
-bool mfxSchedulerCore::SetScheduling(vm_thread& handle)
+bool mfxSchedulerCore::SetScheduling(std::thread& handle)
 {
-    handle;
+    (void)handle;
```
Comment (onabiull) [inline _studio/mfx_lib/scheduler/src/mfx_scheduler_core.cpp]: Why do we need this line?

**Pull request merged**